src/lv_message_efficient.cc

```diff
@@ -215,6 +221,19 @@ namespace grpc_labview
         return protobuf_ptr;
     }
 
+    //---------------------------------------------------------------------
+    //---------------------------------------------------------------------
+    const char* LVMessageEfficient::ParseDouble2DArrayMessage(google::protobuf::uint32 tag, const MessageElementMetadata& fieldInfo, uint32_t index, const char* protobuf_ptr, ParseContext* ctx)
+    {
+        auto metadata = fieldInfo._owner->FindMetadata(fieldInfo.embeddedMessageName);
+        auto nestedMessage = std::make_shared<LVMessage>(metadata);
```

Comment:
Note: Here we create a regular `LVMessage` to copy the type. This is a bit clunky from an efficiency perspective, but I could think of a way to do it without copying into an intermediate message value since there is no guarantee on the order in which fields are sent over the wire.